AcctThread.py

```diff
@@ -0,0 +1,280 @@
+        except KeyError:
+            continue;  # If the field is not found try the next node.
+
+    print ("No more modifying transactions found.  Unexpected stop.\n")
+    print ("Got:")
+    print (json.dumps (jsonMsg, indent = 4))
+    return ""
+
+
+# Write spinner to stderr to show liveness while walking the thread.
+def SpinningCursor ():
+    while True:
+        for cursor in '|/-\\':
+            sys.stderr.write (cursor)
+            sys.stderr.flush ()
+            yield
+            sys.stderr.write ('\b')
+
+
+# Thread the account to extract all transactions performed by that account.
+#    1. Start by getting account_info for the accountId in the most recent
+#       validated ledger.
+#    2. account_info returns the TxId of the last transaction that affected
+#       this account.
+#    3. Call tx with that TxId.  Save that transaction.
+#    4. The tx response should contain the AccountRoot for the accountId.
+#       Extract the new value of PreviousTxID from the AccountRoot.
+#    5. Return to step 3, but using the new PreviousTxID.
+def threadAccount (ws, accountId):
+
+    # Call account_info to get our starting txId.
+    jsonMsg = printAccountInfo (ws, accountId)
+
+    # Handle any errors that might be in the response.
+    if "error" in jsonMsg:
+        print ('No account_info for accountID {0}'.format (accountId))
+        if jsonMsg["error"] == "actMalformed":
+            print ("Did you mistype the accountID?")
+
+        err = jsonMsg["error"]
+        if jsonMsg["error_message"]:
+            err = jsonMsg["error_message"]
+
+        print (err + "  Unexpected stop.")
+        return
+
+    # Extract the starting txId.
+    prevTxnId = ""
+    try:
+        prevTxnId = jsonMsg["result"]["account_data"]["PreviousTxnID"]
+    except KeyError:
+        print (
+            "No PreviousTxnID found for {0}.  No transactions found.".format (
+                accountId))
+        return
+
+    # Transaction threading loop.
+    spinner = SpinningCursor ()  # Liveness indicator.
+    while prevTxnId != "":
+        next (spinner)
+        print ("\n" + ("-" * 79) + "\n")
+        jsonMsg = printTx (ws, prevTxnId)
+        prevTxnId = getPrevTxnId (jsonMsg, accountId)
+
+
+# Open the websocket connection and pass the websocket upstream for use.
+def openConnection (connectTo, accountId):
+    try:
+        ws = create_connection (connectTo)
+    except websocket._exceptions.WebSocketAddressException:
+        print ("Unable to open connection to {0}.\n".format (connectTo))
+        return;
+    except ConnectionRefusedError:
+        print ("Connection to {0} refused.\n".format (connectTo))
+        return
+
+    try:
+        threadAccount (ws, accountId)
+    finally:
+        ws.close ()
+        sys.stderr.write ('\b') # Clean up from SpinningCursor
+
+
+if __name__ == "__main__":
+    # Get command line arguments.
+    connectTo, accountId = extractArgs ()
```

Comment:
These are global variables. Line 34 has a local variable that shadows this global. There's a couple ways to fix this:
1) If you want to keep them global, rename them to use the global convention of `UPPER_CASE`
2) Create a function `main` make the variables local
3) Call the function directly with the tuple returned from `extractArgs`. I.e. `openConnection(*extractArgs())`